src/System.Management.Automation/engine/hostifaces/PSTask.cs

```diff
@@ -449,6 +449,10 @@ public void Start(Runspace runspace)
                     Runspace.DefaultRunspace = runspace;
                     runspace.ExecutionContext.SessionState.Internal.SetLocation(_currentLocationPath);
                 }
+                catch (DriveNotFoundException)
+                {
+                    // Allow task to run if current drive is not available.
+                }
                 finally
                 {
                     Runspace.DefaultRunspace = oldDefaultRunspace;
```

Comment:
I wonder - if we now catch the exception what does the finally?

Reply:
Finally blocks always run, regardless of catch block.

Reply:
I saw that the method behavior was changed with adding the catch but it is an intention of the PR :-)